### PR TITLE
(For Jintae or Gary) Add session level decision.

### PR DESCRIPTION
**Pull request by mjouahri**
*Purpose:* 
- Adds support for the session level decisions.

*Technical overview:*
- Updates data model to reflect the new decision.

**Testing Plan**
- Expanded and added unit tests and run them locally.

**Deployment**
- Publish to Maven
- Github release

### CODE REVIEW

**Comment by jintaekim20** [comment]
Mohammed, can you also update `get_decisions` function and related code to include sessions?

Note: https://siftscience.com/developers/docs/curl/decisions-api/decision-status

File: sift/client.py [hunk 1]
```diff
@@ -502,6 +502,46 @@ def get_order_decisions(self, order_id, timeout=None):
...
 
 
+    def apply_session_decision(self, user_id, session_id, properties, timeout=None):
+        """Apply decision to session
+
+        Args:
+            user_id: id of user
+            session_id: id of session
+            properties:
+                decision_id: decision to apply to user
```
Comment (jintaekim20) [inline sift/client.py]: I guess `decision_id` here is "decision to apply to session" (and that the same comment in `apply_order_decision` should be changed accordingly.)

For example, in the unit test, `decision_id` used was `session_looks_bad_ato`.

File: tests/test_client.py [hunk 2]
```diff
@@ -522,6 +528,42 @@ def test_apply_decision_to_order_ok(self):
             assert(response.http_status_code == 200)
             assert(response.body['entity']['type'] == 'order')
 
+    def test_apply_decision_to_session_ok(self):
```
Comment (jintaekim20) [inline tests/test_client.py]: Just a quick question: no need to test failed cases?

Comment (jintaekim20) [reply tests/test_client.py]: Seems that `test_apply_decision_to_session` does test with no `session_id`. I think it's enough to change the name of `test_apply_decision_to_session`.

File: tests/test_client.py [hunk 3]
```diff
@@ -346,6 +346,52 @@ def test_get_decisions(self):
             assert(response.is_ok())
             assert(response.body['data'][0]['id'] == 'block_user')
 
+    def test_get_decisions_entity_session(self):
+        mock_response = mock.Mock()
+        get_decisions_response_json =  \
+            '{' \
+                '"data": [' \
+                    '{' \
+                        '"id": "block_user",' \
```
Comment (jintaekim20) [inline tests/test_client.py]: well, it's a very minor nit, but `id`, `name`, and `description` do not match the entity type, though it isn't really related to what the test case is supposed to test.

File: tests/test_client.py [hunk 4]
```diff
@@ -346,6 +346,52 @@ def test_get_decisions(self):
...
+                        '"name" : "Block user",' \
+                        '"description": "user has a different billing and shipping addresses",' \
+                        '"entity_type": "session",' \
+                        '"abuse_type": "legacy",' \
```
Comment (jintaekim20) [inline tests/test_client.py]: Also, we only have session decision associated with `account_takeover` today. Its logic may change in the future, and it isn't really related to the test case, either, so you may leave it as is now. It just doesn't look real.

**Comment by garylee1** [comment]
please also update CHANGES.md and the version.py

File: tests/test_client.py (hunk 4 above)
Comment (garylee1) [reply tests/test_client.py]: agree that it would be nice to change this to `account_takeover` to reflect a correct integration pattern

**Comment by garylee1** [comment]
LGTM

**Pull request merged**